lib/api/request/kuzzleRequest.ts

```diff
@@ -710,27 +710,33 @@ export class KuzzleRequest {
   /**
    * Returns the index specified in the request
    */
-  getIndex (): string {
+  getIndex (throwOnMissing = true): string {
```

Comment:
Naming proposal, also it's better to use object named properties for readability but also to be future proof 
```suggestion
  getIndex ({ required = true } = {}): string {
```